clang/lib/CodeGen/CGHLSLRuntime.cpp

```diff
@@ -276,6 +276,29 @@ llvm::Triple::ArchType CGHLSLRuntime::getArch() {
   return CGM.getTarget().getTriple().getArch();
 }
 
+llvm::Intrinsic::ID
+CGHLSLRuntime::getUnsignedIntrinsicVariant(llvm::Intrinsic::ID IID) {
```

Comment:
Rather than having a method to map to the unsigned version of an intrinsic, it might be cleaner to have the macros define the signed and unsigned versions and just have the generated functions return the right intrinsic. That will avoid needing to inline this function in order to avoid the lookup on each generated call.